spyder_kernels/customize/code_runner.py

```diff
@@ -0,0 +1,593 @@
+            yield debug_exec
+        finally:
+            # Reset parent debugger
+            sys.settrace(session.trace_dispatch)
+            session.lastcmd = debugger.lastcmd
+
+            # Reset _previous_step so that get_pdb_state() notifies Spyder about
+            # a changed debugger position. The reset is required because the
+            # recursive debugger might change the position, but the parent
+            # debugger (self) is not aware of this.
+            session._previous_step = None
+
+    @contextmanager
+    def _profile_exec(self):
+        """Get an exec function for profiling."""
+        with tempfile.TemporaryDirectory() as tempdir:
+            # Reset the tracing function in case we are debugging
+            trace_fun = sys.gettrace()
+            sys.settrace(None)
+            # Get a file to save the results
+            profile_filename = os.path.join(tempdir, "profile.prof")
+            try:
+                if self.shell.is_debugging():
+                    def prof_exec(code, glob, loc):
+                        # if we are debugging (tracing), call_tracing is
+                        # necessary for profiling
+                        return sys.call_tracing(cProfile.runctx, (
+                            code, glob, loc, profile_filename
+                        ))
+                    yield prof_exec
+                else:
+                    yield partial(cProfile.runctx, filename=profile_filename)
+            finally:
+                # Resect tracing function
+                sys.settrace(trace_fun)
+                if os.path.isfile(profile_filename):
+                    # Send result to frontend
+                    with open(profile_filename, "br") as f:
+                        profile_result = f.read()
+                    try:
+                        frontend_request(blocking=False).show_profile_file(
+                            profile_result, create_pathlist()
+                        )
+                    except CommError:
+                        logger.debug(
+                            "Could not send profile result to the frontend.")
+
+    def _exec_file(
+        self,
+        filename=None,
+        args=None,
+        wdir=None,
+        post_mortem=False,
+        current_namespace=False,
+        exec_fun=None,
+        canonic_filename=None,
+        context_locals=None,
+        context_globals=None,
+    ):
+        """
+        Execute a file.
+        """
+        if __umr__.enabled:
+            __umr__.run()
+        if args is not None and not isinstance(args, str):
+            raise TypeError("expected a character buffer object")
+
+        try:
+            file_code = self._get_file_code(filename, raise_exception=True)
+        except Exception:
+            print(
+                "This command failed to be executed because an error occurred"
+                " while trying to get the file code from Spyder's"
+                " editor. The error was:\n\n"
```

Comment:
```suggestion
                "This command failed to be executed because an error occurred "
                "while trying to get the file code from Spyder's  editor. "
                "The error was:\n\n"
```